src/store/todo/todo.thunks.ts

```diff
@@ -0,0 +1,128 @@
+import { AnyAction, ThunkAction, ThunkDispatch } from '@reduxjs/toolkit';
+
+import { TodoDAO, TodoDTO } from '~types/todo.types';
+import { AppState } from '~store';
+import apiService from '~services/api.service';
+import { actions } from './todo.actions';
+
+const addTodo =
```

Comment:
I think here it would be also great to move each function to it's own file and then merge them in one. Because in future it's easer to modify one file or delete it then scroll one file with many lines of code.

Reply:
Moved to separate files, merged with index file